echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/EventContent.java

```diff
@@ -31,10 +31,19 @@
   private String executionId;
   private int stageIndex;
 
-  public EventContent(Event event, String type) throws FieldNotFoundException {
+  public EventContent(Event event, String type, String forced_repo, String forced_sha)
+      throws FieldNotFoundException {
     this.event = event;
-    setRepo();
-    setSha();
+    if (!forced_repo.equals("")) {
```

Comment:
not a blocker, but I'm not a big fan of passing an empty string to represent an optional value for `forced_repo` and `forced_sha`. I'd rather mark the parameter `@Nullable` and perform a null check.